pages/blog/[slug].js

```diff
@@ -0,0 +1,54 @@
+import { createClient } from "contentful";
+import Image from "next/image";
+import { documentToReactComponents } from "@contentful/rich-text-react-renderer";
+
+const client = createClient({
+  space: process.env.CONTENTFUL_SPACE_ID,
+  accessToken: process.env.CONTENTFUL_ACCESS_TOKEN,
+});
```

Comment:
This particular code block is being repeated across the app (along with some other Contentful functions). Maybe it would be best to have a file where all Contentful related code exist, allowing any component to import those features instead of implementing them individually.